jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifierSuite.scala

```diff
@@ -308,4 +309,42 @@ class XGBoostClassifierSuite extends FunSuite with PerTest {
     val xgb = new XGBoostClassifier(paramMap)
     xgb.fit(repartitioned)
   }
+
+  test("feature size of predication data should be same with model's") {
```

Comment:
this test name is confusing, what you actually do here is a negative test, instead of a positive one

Reply:
I've moved the test to XGBoostGeneral.suite, and changed the test name to "transform throwing exception when feature size of dataset is different with model's"

Reply:
Since the unit test was copied from https://github.com/dmlc/xgboost/blob/master/jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostGeneralSuite.scala#L349 which I think is a positive case, while the new one I added is a negative case